src/store/modules/app/actions.js

```diff
@@ -1,9 +1,55 @@
 import axios from 'axios'
 import debug from 'debug'
 
+import router from '../../../router'
+
 const logger = debug('app:store:app:actions')
 
+const queryParamsToHandle = {
+  email: {
+    mutationName: 'SET_EMAIL_ADDRESS_TO_CONFIRM',
+  },
+  pendingUserCode: {
+    mutationName: 'SET_PENDING_USER_CODE',
+  },
+
+  // If there's a cached authToken in localstorage this will replace it with
+  //  the one from the query string
+  authToken: {
+    mutationName: 'auth/SET_AUTH_STATE',
+    mutationConfiguration: { root: true },
+  },
+
+  // API error handling
+  errorCode: {
+    mutationName: 'SET_API_ERROR_CODE',
+  },
+  error: {
+    mutationName: 'SET_API_ERROR_MESSAGE',
+  },
+}
+
 export default {
+  HANDLE_QUERY_PARAMS({ commit, rootState }) {
+    logger('HANDLE_QUERY_PARAMS action being executed')
+
+    let paramRemoved = false
+    const { query } = rootState.route
+
+    Object.keys(queryParamsToHandle).forEach((key) => {
+      if (query[key]) {
+        const param = queryParamsToHandle[key]
+
+        commit(param.mutationName, query[key], param.mutationConfiguration)
+        paramRemoved = true
+      }
+    })
+
+    if (paramRemoved) {
+      router.replace(rootState.route.name)
```

Comment:
I don't think there'd be any harm in always calling the replace? Feel free to leave as-is though.

Reply:
Originally I was thinking this would go into an infinite loop but that doesn't appear to be the case (i.e., this only gets called once and isn't invoked again after the first router.replace)